src/main/java/Task.java

```diff
@@ -0,0 +1,141 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Task {
+    private String description;
+    private String date;
+    private String time;
+    private final String symbol;
+    private boolean isAt;
+    private boolean isDone;
+
+    public Task(String description, String date, String time, String symbol, boolean flag) {
+        this.description = description;
+        this.date = date;
+        this.time = time;
+        this.symbol = symbol;
+        this.isAt = flag;
+        this.isDone = false;
+    }
+
+    public String getDate() {
+        return this.date;
+    }
+
+    public String getTime() {
+        return this.time;
+    }
+
+    public void setTime(String time) {
+        this.time = time;
+    }
+
+    /*
+     * Change the done status of a task.
+     *
+     * @return The same task with a changed done status.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /*
+     * Convert the date from "DD/MM/YYYY" format to "D MMM YYYY" format.
+     */
+    public void formatDate() {
+        String copy = this.date;
+        String year = "";
+        String month = "";
+        String day = "";
+        String[] sArr = copy.split("");
+        int slashCounter = 0;
+        for (int i = 0; i < sArr.length; i++) {
+            if (sArr[i].equals("/")) {
+                slashCounter++;
+            } else if (slashCounter == 2) {
+                year += sArr[i];
+            } else if (slashCounter == 1) {
+                month += sArr[i];
+            } else {
+                day += sArr[i];
+            }
+        }
+        if (month.length() == 1) {
+            month = "0" + month;
+        }
+        if (day.length() == 1) {
+            day = "0" + day;
+        }
+        String formattedDate = year + "-" + month + "-" + day;
+        LocalDate ld = LocalDate.parse(formattedDate);
+        formattedDate = ld.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+        this.date = formattedDate;
+    }
+
+    /*
+     * Convert time from 24h format to 12h format.
+     * Separate hour and minute by adding a colon.
+     * Add am or pm depending on time of the day.
+     */
+    public void formatTime() {
+        String copy = this.time;
+        char[] cArr = copy.toCharArray();
+        Integer tensHour = cArr[0] - '0';
+        Integer onesHour = cArr[1] - '0';
+        Integer tensMin = cArr[2] - '0';
+        Integer onesMin = cArr[3] - '0';
+        String formattedTime = "";
+        boolean isAfternoon = false;
+        if (!tensHour.equals(0)) {
+            Integer combinedHour = tensHour * 10 + onesHour;
+            if (combinedHour >= 12) {
+                isAfternoon = true;
+                if (combinedHour >= 13) {
+                    combinedHour -= 12;
+                }
```

Comment:
Should you leverage the LocalDate or some other library to convert the date and time format? I feel that doing so will make the code less cluttered and save you the trouble of writing everything from scratch. This can also be applied to all the classes which inherit from Task.  

Maybe consider something along the lines of this [tutorial](https://howtodoinjava.com/java/date-time/format-time-12-hours-pattern)?

Reply:
Noted, thanks for sharing the link!